internal/sync/syncer.go

```diff
@@ -316,9 +334,12 @@ func (s *Syncer) sync(
 	cluster, err := s.stateCache.GetClusterCache()
 	if err != nil {
 		numaLogger.Error(err, "Error on getting the cluster cache")
+		s.metricsServer.IncKubeCacheFailed()
 		return gitopsSyncCommon.OperationError, err.Error()
 	}
 	openAPISchema := cluster.GetOpenAPISchema()
+	s.metricsServer.SetMonitoredKubeAPI(float64(len(cluster.GetAPIResources())))
+	s.metricsServer.SetKubeCachedResource(float64(len(targetObjs)))
```

Comment:
why this is not increment the number of cached objects? if something is deleted in the cluster. it also needs to be deleted from the cache?